### ISSUE TITLE
Automate Docker Image Release during the packaging build

### ISSUE THREAD

**Issue by dduportal** [jenkins-infra/release#398]
Any Core release should automatically create the (annotated) tag and publish the release in jenkinsci/docker to automate the release process described in the https://github.com/jenkins-infra/release/blob/6a551561776eee535ba6bf8cb65b23be98c1b0b0/docs/releases.md?plain=1#L277-L289 documentation.

This issue is NOT about building the Docker image in release.ci.jenkins.io: it's tracked in https://github.com/jenkins-infra/helpdesk/issues/2845 as a separate topic (because release.ci does not have agents with a Docker CE Engine today).

**Comment by MarkEWaite** [comment]
> Release publication can be done with `gh` CLI:
> 
> * https://cli.github.com/manual/gh_release_edit
> * Something like `gh release edit <new core version> --draft=false --tag=<new core version> --latest`

I think that may need a safeguard so that it does not publish LTS releases or security releases with the same technique.  Weekly releases can take the most recent draft changelog.  LTS and security changelogs are hand-crafted, even on GitHub.

**Comment by timja** [comment]
>  LTS and security changelogs are hand-crafted,

LTS ones follow a very strong pattern though and could just have a template supplied, if someone wants to polish it they can do that after, better to minimise human interaction required delaying the docker release etc.

If it's important it's not update then it could just do a tag and release notes added later.

**Comment by MarkEWaite** [comment]
> > LTS and security changelogs are hand-crafted,
> 
> LTS ones follow a very strong pattern though and could just have a template supplied, if someone wants to polish it they can do that after, better to minimise human interaction required delaying the docker release etc.

Agreed.  It can be automated for LTS releases as you said.  They follow a very consistent pattern.

**Comment by dduportal** [comment]
Scheduling constraint we met during yesterday weekly release (2.414):

The tag on jenkinsci/docker must be created only after the WAR file is available in get.jenkins.io to avoid the `curl: (22) The requested URL returned error: 404` error message in the Docker image build (when downloading the aformentionned WAR file).
